src/plugin/class-subjects-controller.php

```diff
@@ -220,15 +64,33 @@ public function init_schema(): void {
 					'transformedId' => array(
 						'description' => __( 'Post ID of transformed result of this liberated page', 'try_wordpress' ),
 						'type'        => 'string',
-						'context'     => array( 'view', 'edit' ),
 						'required'    => false,
 						'arg_options' => array(
 							'sanitize_callback' => 'sanitize_text_field',
 						),
 					),
 				),
-			),
-		);
+			);
+
+			foreach ( $definition['fields'] as $field => $field_definition ) {
+				$this->schema[ $subject_type ]['properties'][ 'raw' . ucfirst( $field ) ]    = array(
+					'description' => '[raw]' . $field_definition['description'] ?? '',
+					'type'        => convert_schema_type_to_rest_api_type( $field_definition['type'] ),
```

Comment:
I think there are two concepts, that represent two different things:

1. The type of field from the schema (`data`, `text`, `html`)
2. The type of data of the property of the REST API. In our case this will always be `string`.

I think that the `type` property (on line 78) relates to 2. and not 1. right?

If so, I think we could hardcoded it as `string`, which would mean we no longer need the `convert_schema_type_to_rest_api_type()` function.

I might be misinterpreting something, please correct me if I'm wrong.

Reply:
REST API allows 'array', 'object', 'string', 'number', 'integer', 'boolean', null

So we only want to replace a type value from our schema which doesn't match with either of the options. While `string` is a good default, if we ever find ourselves specifying a `number` for a particular field, we want it to be a number or integer and not just be treated as string because of that choice, which is why I am explicitly mapping them here.

Reply:
Sounds reasonable, thanks for explaining.